### PR TITLE
feat(defintions): allow setting of scope

### LINKED ISSUE #170: [RFC]: How to handle scope of @definition.function
Things like `goto_definition` are based that on the assumption that all `@definition`s have a surrounding scope in which they are valid. But there is a problem: Most `locals.scm` define functions as separate scopes which makes sense since all the arguments of a function are only valid in this function. However, the name of the function is defined for the surrounding scope not the scope defined by the function itself.

How can we tackle that problem?

### PR DESCRIPTION
**Pull request by steelsojka**
This should fix #170 

I still have to go and update the language files for all applicable use cases.

### CODE REVIEW

File: lua/nvim-treesitter/locals.lua [hunk 1]
```diff
@@ -172,6 +174,40 @@ M.get_definitions_lookup_table = ts_utils.memoize_by_buf_tick(function(bufnr)
   return result
 end)
 
+--- Gets all the scopes of a definition based on the scope type
+-- Scope types can be
+--
+-- "parent": Uses the parent of the containing scope, basically, skipping a scope
+-- "global": Uses the top most scope
+-- nil: Uses the containg scope of the definition
```
Comment (stsewd) [inline lua/nvim-treesitter/locals.lua:182]: These should be documented in contributing.md

Comment (steelsojka) [reply lua/nvim-treesitter/locals.lua:182]: done

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -177,3 +177,29 @@ Mainly for markup languages.
...
+```javascript
+function doSomething() {}
+
+doSomething(); // Should point to the declaration as the definition
+```
+
+```scheme
+(function_declaration
+  ((identifier) @definition.var)
+   (set! "definition.var.scope" "parent"))
```
Comment (stsewd) [inline CONTRIBUTING.md:198]:
```suggestion
   (#set! "definition.var.scope" "parent"))
```

File: queries/javascript/locals.scm [hunk 3]
```diff
@@ -43,6 +43,10 @@
 (import_specifier
   (identifier) @definition.import)
 
+(function_declaration
+  ((identifier) @definition.var)
+   (set! "definition.var.scope" "parent"))
```
Comment (stsewd) [inline queries/javascript/locals.scm:48]:
```suggestion
   (#set! "definition.var.scope" "parent"))
```

File: lua/nvim-treesitter/locals.lua (hunk 1 above)
Comment (stsewd) [inline lua/nvim-treesitter/locals.lua:182]: I think we may need to have `local` as an option too. This is, nil explicitly (we don't have something like unset or set to nil I guess).

Comment (stsewd) [reply lua/nvim-treesitter/locals.lua:182]: This would be useful to "reset" the value in another query.

**Comment by theHamsta** [comment]
typo in commit message

File: lua/nvim-treesitter/locals.lua (hunk 1 above)
Comment (steelsojka) [reply lua/nvim-treesitter/locals.lua]: done :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (stsewd) [reply CONTRIBUTING.md:198]: Looks like this wasn't updated

**Pull request merged**